test/sha256_test.cpp

```diff
@@ -64,4 +66,80 @@ TEST_P(Sha256Test, IUF) {
 
 INSTANTIATE_TEST_SUITE_P(Sha256Regression,
                          Sha256Test,
-                         testing::ValuesIn(g_Cases));
\ No newline at end of file
+                         testing::ValuesIn(g_Cases));
+
+TEST(Sha256Test, uint256_compareTest) {
+  VeriBlock::uint256 hash1(
+      "00000000000000000000000000000000000000000000000000000000000000ff"_unhex);
+  VeriBlock::uint256 hash2(
+      "0000000000000000000000000000000000000000000000000000000000000fff"_unhex);
+
+  ASSERT_TRUE(hash2 > hash1);
+  ASSERT_TRUE(hash1 < hash2);
+
+  hash1 = VeriBlock::uint256(
+      "00000000000000000000000000000000000000000000000000000000000033ff"_unhex);
+  hash2 = VeriBlock::uint256(
+      "00000000000000000000000000000000000000000000000000000000000032ff"_unhex);
+
+  ASSERT_TRUE(hash1 > hash2);
+  ASSERT_TRUE(hash2 < hash1);
+
+  hash1 = VeriBlock::uint256(
+      "fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff322"_unhex);
+  hash2 = VeriBlock::uint256(
+      "fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff322"_unhex);
+
+  ASSERT_TRUE(hash1 == hash2);
+}
+
+TEST(Sha256Test, uint256_decodeBits) {
+  uint256 target = decodeBits(0x04800000);
+
+  EXPECT_EQ(
+      target,
+      uint256(
+          "0000000000000000000000000000000000000000000000000000000000000000"_unhex));
```

Comment:
pls rewrite this test to be value-parametrized https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests

similar to vblake_test or sha256_test

```
struct TestCase {
  uint32_t bits;
  uint256 target;
}
```